test/rotation.jl

```diff
@@ -0,0 +1,80 @@
+function rotation_testsuite(Backend)
+    device(x) = adapt(Backend(), x)
+    gradtest_fn = Backend == CPU ? gradtest : gputest
+    T = Float32
+    atol = T == Float32 ? 1e-3 : 1e-6
+    rtol = T == Float32 ? 1f-2 : 1f-6
+    angles = deg2rad.([0, 0.0, 0.0001, 35, 90, -90, -90.0123, 170, 180, 270, 360, 450, 1234.1234]) 
+
+    @testset "imrotate" begin
+        @testset "Simple test" begin
+            arr = device(zeros((6, 6, 1, 1))); 
+            arr[3:4, 4, 1, 1] .= 1;
+            @test all(cpu(NNlib.imrotate(arr, deg2rad(45))) .≈ [0.0 0.0 0.0 0.0 0.0 0.0; 0.0 0.0 0.0 0.0 0.0 0.0; 0.0 0.0 0.0 0.29289321881345254 0.585786437626905 0.0; 0.0 0.0 0.08578643762690495 1.0 0.2928932188134524 0.0; 0.0 0.0 0.0 0.08578643762690495 0.0 0.0; 0.0 0.0 0.0 0.0 0.0 0.0])
+        end
+
+
+        @testset "Compare with ImageTransformations" begin
+            for sz in [(51,51,1,1)]
+                
+                arr1 = device(zeros(T, sz))
+                arr1[15:40, 15:40, :, :] .= device(1 .+ randn((26, 26)))                                                                       
+                arr2 = device(zeros(T, (sz[1], sz[2], sz[3], 3)))
+                arr2[15:40, 15:40, :, :] .= device(arr1[15:40, 15:40, :, :])
+
+                for method in [:nearest, :bilinear]
+                    @testset "$method" begin
+                        for angle in angles
+                            if iseven(sz[1])
+                                res1 = cpu(NNlib.imrotate(arr1, angle; method, midpoint=size(arr1).÷2 .+ 0.5))
+                                res2 = cpu(NNlib.imrotate(arr2, angle; method, midpoint=size(arr1).÷2 .+ 0.5))
+                            else
+                                res1 = cpu(NNlib.imrotate(arr1, angle; method))
+                                res2 = cpu(NNlib.imrotate(arr2, angle; method))
+                            end
+                            if method == :nearest
+                                res_IT = ImageTransformations.imrotate(cpu(arr1)[:, :, 1, 1], angle, axes(arr1)[1:2], method=Constant(), fillvalue=0)
+                            elseif method == :bilinear
+                                res_IT = ImageTransformations.imrotate(cpu(arr1)[:, :, 1, 1], angle, axes(arr1)[1:2], fillvalue=0)
+                            end
+                            @show res1, res_IT 
+                            @test all(.≈(1 .+ res1[:, :, :, :], 1 .+ res_IT[:, :], rtol=rtol))
```

Comment:
Why do you add 1 here?

Reply:
because of the zeros. Sometimes zero comparisons fail (didn't check in this case)

Reply:
ok, just keep it then :)